webapp/sources/rudder/rudder-core/src/test/scala/com/normation/rudder/db/DBCommon.scala

```diff
@@ -79,24 +79,26 @@ trait DBCommon extends Specification with Loggable with BeforeAfterAll {
    * everything is temporary
    */
   def sqlInit: String = {
-    val is      = this.getClass().getClassLoader().getResourceAsStream("reportsSchema.sql")
-    val sqlText = Source
+    val is              = this.getClass().getClassLoader().getResourceAsStream("reportsSchema.sql")
+    // This is a way to create and use all DDL under a schema, that will be dropped at the end of tests
+    val setTestNamepace = {
+      """
+        |CREATE SCHEMA temp_schema;
+        |SET search_path TO temp_schema;
```

Comment:
what happens if tests are interrupted and the `DROP SCHEMA temp_schema CASCADE` is never done ? Do we hand up with that schema being used on that database by default, even in new connections? 
If so, is there a way to say "SET search_path TO temp_schema" but only for current connection (and so revert to the default schema in any case once the connection is terminated ?)

Reply:
Yes the schema would still remain after tests have executed.
Provided that we can create a schema-per-connection, we should indeed do as you say. 
And it seems to be the case (https://dba.stackexchange.com/a/76661), so I changed it and it is even simpler : https://github.com/Normation/rudder/pull/5813/commits/5193ce3116f7d003ef4b29e037b2171840e95645